### PR TITLE
ADEN-9619 Major changes in tracking-opt-in implementation

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

**Comment by jacekbj** [comment]
Are you aware that `package-lock.json` has been updated? Just checking, I'm not against updating dependencies but that was not strictly required since nothing in package.json changed.

**Comment by dmnsobczak** [comment]
Yes, I am. I had some problems with compiling the library - it's a bit ancient now. Removing the node_modules and package-lock fixed the issue for me so I decided to commit it after rewrite.
@rybmat Can you please add third opinion if it's safe?

**Review by jacekbj — APPROVED** [review]

File: src/tracking/tracking-queue.js [hunk 1]
```diff
@@ -41,10 +45,11 @@
...
 	};
 
-	M.geoRequiresConsent = instance.geoRequiresTrackingConsent();
+	M.geoRequiresConsent = instances.gdpr.geoRequiresTrackingConsent();
+	M.geoRequiresSignal = instances.ccpa.geoRequiresUserSignal();
 	M.resetTrackingOptIn = function () {
-		instance.reset();
+		instances.gdpr.reset();
 	};
 	// TODO: Remove this flag once we fully switch to CMP from TrackingOptIn - ADEN-7432
```
Comment (fraszczakszymon) [inline src/tracking/tracking-queue.js]: This todo can be addressed now, what do you think?

**Review by bckatarzyna — APPROVED** [review]

File: src/tracking/tracking-queue.js (hunk 1 above)
Comment (dmnsobczak) [reply src/tracking/tracking-queue.js]: Yep, line removed.

**Pull request merged**